File: TwoPointers/2109_Adding_Spaces_to_a_String/2109_Adding_Spaces_to_a_String.cpp
class Solution {
public:
    string addSpaces(string s, vector<int>& spaces) {
        string res = "";
        size_t n = s.length();
        size_t j = 0;
        for(size_t i=0; i<n; i++) {
            if(j<spaces.size() && i==spaces[j]) {
                res.push_back(' ');
                j++;
            }
            res.push_back(s[i]);
        }

        return res;
    }
};

File: TwoPointers/2109_Adding_Spaces_to_a_String/2109_Adding_Spaces_to_a_String.java
class Solution {
    public String addSpaces(String s, int[] spaces) {
        StringBuilder res = new StringBuilder();    // need StringBuilder to append to string efficiently, else Time Limit Exceeded
        int j = 0;
        int n = s.length();
        for(int i=0; i<n; i++) {
            if(j<spaces.length && i==spaces[j]) {
                res.append(' ');
                j++;
            }
            res.append(s.charAt(i));
        }
        return res.toString();
    }
}